src/main/java/org/hydev/mcpm/utils/ZstdUtils.java

```diff
@@ -0,0 +1,170 @@
+package org.hydev.mcpm.utils;
+
+import io.airlift.compress.MalformedInputException;
+import io.airlift.compress.zstd.ZstdCompressor;
+import io.airlift.compress.zstd.ZstdDecompressor;
+import org.jetbrains.annotations.Nullable;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+
+/**
+ * Utility for Zstd-jni
+ *
+ * @author Azalea (https://github.com/hykilpikonna)
+ * @since 2022-11-22
+ */
+public class ZstdUtils
+{
+    public static final ReflectedZstd zstd;
+
+    /**
+     * Reflected zstd class
+     *
+     * @param cl Class object
+     * @param decompress decompress(byte[], int) method
+     * @param decompressedSize decompressedSize(byte[]) method
+     * @param compress compress(byte[]) method
+     */
+    public record ReflectedZstd(Class<?> cl, Method decompress, Method decompressedSize, Method compress) { }
+
+    /**
+     * Get zstd class
+     *
+     * @return Zstd class if usable, null otherwise
+     */
+    private static @Nullable ReflectedZstd getZstdClass()
+    {
+        try
+        {
+            var cl = Class.forName("com.github.luben.zstd.Zstd");
+            var decompress = cl.getMethod("decompress", byte[].class, int.class);
+            var decompressedSize = cl.getMethod("decompressedSize", byte[].class);
+            var compress = cl.getMethod("compress", byte[].class, int.class);
+            return new ReflectedZstd(cl, decompress, decompressedSize, compress);
+        }
+        catch (Throwable e)
+        {
+            // Any exception here would mean that it isn't supported
+            return null;
+        }
+    }
+
+    static
+    {
+        zstd = getZstdClass();
```

Comment:
Wish this was truly lazy but I don't really care that much.

Reply:
Just 21 ms so it's fine

<img width="240" alt="image" src="https://user-images.githubusercontent.com/22280294/203689839-ff369d82-b777-4a9b-aa56-847993a49b93.png">